src/Rings/MPolyMap/Types.jl

```diff
@@ -33,17 +33,14 @@ const _DomainTypes = Union{MPolyRing, MPolyQuoRing}
     # argument to the outer constructors or make the outer constructors 
     # call the inner one with this argument set to `false`. This way the check 
     # can safely be disabled.
-    if check_for_mapping_of_vars && all(_is_gen, img_gens) && _allunique(img_gens)
-      gens_codomain = gens(codomain)
-      result.variable_indices = [findfirst(_cmp_reps(x), gens_codomain) for x in img_gens]
+    if check_for_mapping_of_vars
+      result.variable_indices = __maps_variables_to_variables(img_gens,
+                                                              codomain)
```

Comment:
In the previous implementation this field being unassigned was the indicator that the map does not take variables into variables. Now it seems that the field is assigned by default. But I suppose you deliberately changed the overall concept here and took that into account, right? 

Either way: I was just a bit puzzled that this field is assigned only if a specific kwarg is set to true.

Reply:
1. The part with always setting the field is on purpose. Repeatedly constructing `Int[]` is expensive. Now one should never query `isdefined(f, :)` directly, but just ask whether `_maps_...` is true.
2. The special kwarg was added by you. I am happy to remove it, since it is never used and cannot be used legally using `hom`, see https://github.com/oscar-system/Oscar.jl/pull/4124#issuecomment-2383883603.

Reply:
I see. I think at the time I wanted to leave some backdoor open to avoid the check. That's why I introduced the kwarg, but did not yet forward it to the user-facing constructors. If it does not conflict with your implementation, then I would say: let's just keep it. 

Not relying on `isdefined` anymore makes sense. This was a bit opaque, anyway.

Reply:
OK, let's keep keeping it. Thanks for the review.